### PR TITLE
feat: add streaming checklists

### PR DESCRIPTION
**Pull request by mariobuikhuizen**

### CODE REVIEW

File: nominal/core/checklist.py [hunk 1]
```diff
@@ -170,6 +175,49 @@ def _from_conjure(cls, clients: _Clients, checklist: scout_checks_api.VersionedC
...
+        assets: Sequence[Asset | str],
+        notification_configurations: Sequence[str],
+        *,
+        evaluation_delay: timedelta = timedelta(),
+        recovery_delay: timedelta = timedelta(seconds=15),
+    ) -> None:
+        """Execute the checklist for the given assets.
+        - `assets`: Can be `Asset` instances, or Asset RIDs.
+        - `notification_configurations`: Integration RIDs, checklist violations will be sent to the specified
+           integrations.
```
Comment (ConnorDang0) [inline nominal/core/checklist.py:189]: Maybe add here that we require there to be at least 1 notifactionConfiguration provided.

**Review by ConnorDang0 — COMMENTED** [review]
Can we also add an implementation for hitting the `reloadStreamingChecklist` endpoint?

**Comment by mariobuikhuizen** [comment]
This is blocked until there is a way to link a Checklist to an Asset

File: nominal/core/checklist.py [hunk 2]
```diff
@@ -170,6 +175,53 @@ def _from_conjure(cls, clients: _Clients, checklist: scout_checks_api.VersionedC
...
+        evaluation_delay: timedelta = timedelta(),
+        recovery_delay: timedelta = timedelta(seconds=15),
+    ) -> None:
+        """Execute the checklist for the given assets.
+        - `assets`: Can be `Asset` instances, or Asset RIDs.
+        - `notification_configurations`: Integration RIDs, checklist violations will be sent to the specified
+           integrations. At least one integration must be specified.
+        - `evaluation_delay`: Delays the evaluation of the streaming checklist. This is useful for when data is delayed.
+        - `recovery_delay`: Specifies the minimum amount of time that must pass before a check can recover from a
+                            failure. Minimum value is 15 seconds.
```
[the remark this answers is not in the document]
Comment (mariobuikhuizen) [reply nominal/core/checklist.py:192]: I copied that from the docstring of the Conjure property.

File: nominal/core/checklist.py [hunk 3]
```diff
@@ -170,6 +175,53 @@ def _from_conjure(cls, clients: _Clients, checklist: scout_checks_api.VersionedC
...
+        assets: Sequence[Asset | str],
+        notification_configurations: Sequence[str],
+        *,
+        evaluation_delay: timedelta = timedelta(),
+        recovery_delay: timedelta = timedelta(seconds=15),
+    ) -> None:
+        """Execute the checklist for the given assets.
+        - `assets`: Can be `Asset` instances, or Asset RIDs.
+        - `notification_configurations`: Integration RIDs, checklist violations will be sent to the specified
+           integrations. At least one integration must be specified.
```
[the remark this answers is not in the document]
Comment (ConnorDang0) [reply nominal/core/checklist.py]: Currently, `NotificationConfiguration` is an object with only 1 field `integrationRid`. In the future, we may add more fields to surface most customization options for the notification (e.g. message and tags).

**Review by jackparmer — APPROVED** [review]

**Pull request merged**